pydantic/main.py

```diff
@@ -418,9 +433,25 @@ def __iter__(self) -> 'AnyGenerator':
         """
         yield from self._iter()
 
-    def _iter(self, by_alias: bool = False) -> 'TupleGenerator':
+    def _iter(self, by_alias: bool = False, skip_defaults: bool = True) -> 'TupleGenerator':
         for k, v in self.__values__.items():
-            yield k, self._get_value(v, by_alias=by_alias)
+            yield k, self._get_value(v, by_alias=by_alias, skip_defaults=skip_defaults)
+
+    def _calculate_keys(
+        self, include: Set[str] = None, exclude: Optional[Set[str]] = set(), skip_defaults: bool = False
+    ) -> Set[str]:
+        if skip_defaults:
+            keys = self.__fields_set__.copy()
+        else:
+            keys = set(self.__values__.keys())
```

Comment:
I doubt we need `set()` here.

Reply:
If we do not convert it to a set (there is no copy here). We rely on the fact that dict keys are special sets which return LHS references if modified in place. Seems a bit ugly, when we can explicitly copy the data here and move on with things.

Reply:
ok, that's fine.